vocode/streaming/synthesizer/deepgram_synthesizer.py

```diff
@@ -0,0 +1,82 @@
+import asyncio
+from typing import Optional
+
+from deepgram import DeepgramClient, DeepgramClientOptions, SpeakOptions
+
+from vocode.streaming.models.audio import AudioEncoding, SamplingRate
+from vocode.streaming.models.message import BaseMessage
+from vocode.streaming.models.synthesizer import DeepgramSynthesizerConfig
+from vocode.streaming.synthesizer.base_synthesizer import BaseSynthesizer, SynthesisResult
+from vocode.streaming.utils.create_task import asyncio_create_task_with_done_error_log
+from vocode.streaming.utils.mp3_helper import decode_mp3
+
+
+class DeepgramSynthesizer(BaseSynthesizer[DeepgramSynthesizerConfig]):
+    def __init__(
+        self,
+        synthesizer_config: DeepgramSynthesizerConfig,
+        api_key: Optional[str] = "",
+    ):
+        super().__init__(synthesizer_config)
+
+        # save config
+        self.synthesizer_config = synthesizer_config
+
+        self._config: SpeakOptions = SpeakOptions()
```

Comment:
could we use `pcm` as the default audio encoding here? should avoid the mp3 conversion and will handle mulaw

Reply:
Will update this 👍